src/main/java/com/amazon/jenkins/ec2fleet/EC2RetentionStrategy.java

```diff
@@ -62,6 +70,16 @@ public long check(final EC2FleetNodeComputer fc) {
                 //  Hence, determine the reasons for termination here using persisted fields for accurate handling of termination.
                 if (fc.isMarkedForDeletion()) {
                     reason = EC2AgentTerminationReason.AGENT_DELETED;
+                } else if (isOrphaned) {
+                    reason = EC2AgentTerminationReason.AGENT_ORPHANED;
+                    try {
+                        //TODO: terminate instance and move this logic into a method
+                        Jenkins.get().removeNode(node);
```

Comment:
A design decision made a while ago to schedule instance for termination rather than terminating them directly. For the sake of consistency, lets stick to that. Also, you will need to add this reason to (master) L#77.

Was there a reason for this line here instead of just marking the reason and letting (master) L#78-83 below handle the termination logic?

Reply:
Unfortunately based on the logic we have right now, the cloud cannot be null if we want to terminate a node. In this case we would be unable to ever add the instance to the scheduled termination list. If this solution is unacceptable we may need to find a different design for handling nodes to terminate and eventually instances.

Reply:
duh! I missed this. Per our offline sync, might be best to handle orphaned nodes later after the release for cloud tracking goes out.